cypress/e2e/po/components/base-page.po.ts

```diff
@@ -1,30 +1,52 @@
-import ComponentPo, { GetOptions } from '@/cypress/e2e/po/components/component.po';
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import type { GetOptions } from '@/cypress/e2e/po/components/component.po';
 import ResourceTablePo from '@/cypress/e2e/po/components/resource-table.po';
 import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
 import ResourceDetailPo from '@/cypress/e2e/po/edit/resource-detail.po';
 
-export class SharedComponentsPo extends ComponentPo {
-  resourceTable(options?: GetOptions) {
-    return new ResourceTablePo(this.self(options));
+export abstract class BasePagePo extends PagePo {
+/**
+ * Returns resource detail page object
+ * @param options
+ * @returns
+ */
+  resourceDetail(options?: GetOptions) {
+    return new ResourceDetailPo(this.self(options));
```

Comment:
ResourceDetailPo's ctor is from ComponetPo which should be a selector for the component.